dotnet-roslyn/analyzers/Audacia.CodeAnalysis.Analyzers/src/Audacia.CodeAnalysis.Analyzers/Rules/MethodLength/MethodLengthAnalyzer.cs

```diff
@@ -156,7 +156,25 @@ public override void Visit(SyntaxNode node)
 
             private bool IsStatement(SyntaxNode node)
             {
-                return !node.IsMissing && node is StatementSyntax statement && !IsExcludedStatement(statement);
+                return !node.IsMissing && node is StatementSyntax statement && !IsExcludedStatement(statement) && !IsLoggingStatement(node);;
+            }
+            
+            private static bool IsLoggingStatement(SyntaxNode node)
+            {
+                // Check if the node is an invocation expression
+                if (node is ExpressionStatementSyntax expressionStatement &&
+                    expressionStatement.Expression is InvocationExpressionSyntax invocation)
+                {
+                    // Check if the expression being invoked is a member access (like "logger.Info")
+                    if (invocation.Expression is MemberAccessExpressionSyntax memberAccess)
+                    {
+                        // Check if the name of the method being accessed starts with "Log"
+                        return memberAccess.Name.Identifier.Text.StartsWith("Log");
```

Comment:
question: is it possible to make this use the actual namespace of the logger? I wouldn't want any other method names that happen to start with `Log` to be excluded

Reply:
I did think about this, what would you expect to see as the namespace?

Reply:
Presumably it would be a check that the Log method exists on `ILogger` from `Microsoft.Extensions.Logging`. That would assume everybody is using that for logging - unless there is a way to may it configurable?

https://learn.microsoft.com/en-us/dotnet/api/microsoft.extensions.logging.ilogger?view=dotnet-plat-ext-8.0

Reply:
Agreed as I'm guessing 90% of our projects use `Microsoft.Extensions.Logging` but I was hesitant to add this additional check as I wasn't sure on whether we wanted that restriction for projects that used `NLog`, `Serilog` or any other logging library that we may decide to use in the future.

TLDR: Happy to add as long as we all agree it's the correct decision

Reply:
Regardless of the logging provider, I would assume you'd still be injecting a `Microsoft.Extensions.Logging.ILogger` and calling `_logger.LogBlah`, so I'm happy that this would be useful for the vast majority of projects.

Might be worth looping in @richardb355 at this stage though.

Reply:
I agree with @OwenLacey28 that, even if someone is using something like Serilog, they should still be using the `ILogger` type from `Microsoft.Extensions.Logging`.